lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -121,6 +121,29 @@ def ks_console
             (@port && @baud) ? "console=ttyS#{@port},#{@baud}" : ''
           end
 
+          apipie :method, 'Returns manage_packages snippet with packages & action' do
+            required :action, String, "Allowed actions: 'install', 'update', 'remove', 'group install' and 'group remove'"
```

Comment:
`apt update` and `yum update` are entirely different commands. The implementation suggests you have implemented things how yum does it, but as a user it would surprise me. `group` commands are not something `apt` knows either. This needs more documentation to describe what it does.

Reply:
> apt update and yum update are entirely different commands. The implementation suggests you have implemented things how yum does it, but as a user it would surprise me.

Good point, this needs to be done differently.

> group commands are not something apt knows either. 

It's taken care of in the snippet.

> This needs more documentation to describe what it does.

Well the logic is in the snippet. This macro only returns the snippet, with some small logic to it.
If users wants to know what is happening, snippet is right place to go.

Reply:
> Well the logic is in the snippet. This macro only returns the snippet, with some small logic to it.
> If users wants to know what is happening, snippet is right place to go.

Then the documentation should say that. Right now I think the documentation isn't very clear in that. I still question whether we should even have macros that wrap snippets in the first place. Until now we've always used the snippet tag inside templates when we wanted to reuse logic.